src/components/LayerSwitcher/AddCustomLayer.tsx

```diff
@@ -0,0 +1,283 @@
+import * as React from 'react';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import { Button, CircularProgress, TextField } from '@material-ui/core';
+import { loadLayers } from '@dlurak/editor-layer-index';
```

Comment:
1) Can we please move this function into helpers folder of osmapp? It looks like `loadLayers()` is only a wrapper around `fetch(imagery.json)`, i would much prefer to use our osmapp `fetchJson`, which does some caching, is abortable etc. 

2) Their README.md mentions this url:`https://osmlab.github.io/editor-layer-index/imagery.geojson`

3) Also I prefer types written as typescript types, could you please use them? 
eg (written by chatgpt, only for idea)
```
interface Attribution {
    required: boolean;
    text: string;
    url: string;
}

interface Extent {
    max_zoom?: number;
        min_zoom?: number;
    polygon: [number, number][][];
}

interface Imagery {
    attribution: Attribution;
    country_code: string;
    description: string;
    end_date: string;
    extent: Extent;
    icon: string;
    id: string;
    license_url: string;
    name: string;
    privacy_policy_url: string;
    start_date: string;
    type: string;
    url: string;
        best?: boolean;
    min_zoom?: number;
    start_date?: string;
    end_date?: string;
    switch?: string;

}
```